File: baselibrary/src/main/java/com/anningtex/baselibrary/base/BaseContract.java
package com.anningtex.baselibrary.base;

public interface BaseContract {
    interface View {
        void showLoading();

        void dismissLoading();

        void toast(String msg);

        void toast(int msg);
    }

    interface Presenter<V extends View> {
        void attachView(V view);

        void detachView();
    }
}

File: baselibrary/src/main/java/com/anningtex/baselibrary/base/BaseContract.kt
package com.anningtex.baselibrary.base

interface BaseContract {
    interface View {
        fun showLoading()
        fun dismissLoading()
        fun toast(msg: Int)
        fun toast(msg: String)
    }

    interface Presenter<V : View?> {
        fun attachView(view: V)
        fun detachView()
    }
}